src/output-json-tls.c

```diff
@@ -119,30 +155,145 @@ void JsonTlsLogJSONExtended(json_t *tjs, SSLState * state)
             break;
         default:
             snprintf(ssl_version, SSL_VERSION_LENGTH, "0x%04x",
-                     state->server_connp.version);
+                     ssl_state->server_connp.version);
             break;
     }
-    json_object_set_new(tjs, "version", json_string(ssl_version));
+    json_object_set_new(js, "version", json_string(ssl_version));
+}
 
-    /* tls.notbefore */
-    if (state->server_connp.cert0_not_before != 0) {
+static void JsonTlsLogNotBefore(json_t *js, SSLState *ssl_state)
+{
+    if (ssl_state->server_connp.cert0_not_before != 0) {
         char timebuf[64];
         struct timeval tv;
-        tv.tv_sec = state->server_connp.cert0_not_before;
+        tv.tv_sec = ssl_state->server_connp.cert0_not_before;
         tv.tv_usec = 0;
         CreateUtcIsoTimeString(&tv, timebuf, sizeof(timebuf));
-        json_object_set_new(tjs, "notbefore", json_string(timebuf));
+        json_object_set_new(js, "notbefore", json_string(timebuf));
     }
+}
 
-    /* tls.notafter */
-    if (state->server_connp.cert0_not_after != 0) {
+static void JsonTlsLogNotAfter(json_t *js, SSLState *ssl_state)
+{
+    if (ssl_state->server_connp.cert0_not_after != 0) {
         char timebuf[64];
         struct timeval tv;
-        tv.tv_sec = state->server_connp.cert0_not_after;
+        tv.tv_sec = ssl_state->server_connp.cert0_not_after;
         tv.tv_usec = 0;
         CreateUtcIsoTimeString(&tv, timebuf, sizeof(timebuf));
-       json_object_set_new(tjs, "notafter", json_string(timebuf));
+       json_object_set_new(js, "notafter", json_string(timebuf));
+    }
+}
+
+static void JsonTlsLogCertificate(json_t *js, SSLState *ssl_state)
+{
+    if ((ssl_state->server_connp.cert_input == NULL) ||
+            (ssl_state->server_connp.cert_input_len == 0)) {
+        return;
+    }
+
+    SSLCertsChain *cert = TAILQ_FIRST(&ssl_state->server_connp.certs);
+    if (cert == NULL) {
+        return;
+    }
+
+    unsigned long len = cert->cert_len * 2;
+    uint8_t encoded[len];
+    Base64Encode(cert->cert_data, cert->cert_len, encoded, &len);
```

Comment:
Please check retval like in https://github.com/inliniac/suricata/blob/master/src/output-json-alert.c#L399

It's missing in a few other places both in your PR (and in master too). We should probably make Base64Encode with WARN_UNUSED